spec/fastsync/vdd/fastsyncL2.tla

```diff
@@ -0,0 +1,528 @@
+- BlockReply
+ 
+Correct peers are part of the replicated state machine, i.e., they manage
+blockchain and execute transactions against the same deterministic application A.
+We assume Tendermint failure model, i.e., given a trusted header h, we assume that >2/3 of voting power will behave
+correctly until time h.Time + TRUSTED_PERIOD. We don't make any assumptions about behavior of faulty processes.
+
+In this spec we assume a simpler problem (compared to complete fast-sync) defined with the following assumptions: 
+
+- node is initialised with a set of peers with which messages can be exchanged
+- peers can be removed (in case they misbehave) but they cannot be added
+- some numbers of peers is faulty (correct and faulty peer ids are input parameter)
+- peers do not modify its height, i.e., we take a snaphsot of the system and try to sync as much as we can
+within that shapshot; therefore StatusRequest message is sent only once
+- we assume synchronous communication between correct processes (node and correct peers) and model communication
+with asycnhronous rpc calls, i.e, every request receives corresponding reply 
+- faulty processes can send arbitrary messages (even if they are not requested)
+- fast sync should terminate with at least a block height that is equal to the maximum height of the correct peer
+
+This spec can be extended allong the following lines to capture full fast sync problem:
+
+1) Node can establish connections to other peers during fast sync protocol. This should probably be modelled by having
+additional input message received by Node (AddPeer). This should be relatively easily supported
+as it shouldn't significantly affect termination conditions. More preciselly, the properties should be very similar 
+in case we bound eventually reception of AddPeer messages; otherwise we can't terminate as AddPeer message can 
+always delay termination.
+
+2) Peers modify its height (blockchain grows) and node wants to try to sync to the latest block. In this case, node
+needs to send multiple StatusRequest messages. We might want to try to model this variant as multi-round version of this
+spec where round corresponds to a single loop (StatusRequest sent -> All corresponding StatusResponses received -> Termination 
+condition satisfied). In this case starting next round could be modelled as an external event (NextRound). 
+Termination condition would then be constrained of eventually stopping to receive NextRound events.   
+*)
+
+EXTENDS Integers, FiniteSets, Sequences 
+
+CONSTANTS MAX_HEIGHT, 
+          CORRECT, 
+          FAULTY, 
+          TARGET_PENDING, 
+          PEER_MAX_REQUESTS
+
+\* the set of potential heights
+Heights == 1..MAX_HEIGHT
+
+\* simplifies execute blocks logic. Used only in block store. 
+HeightsPlus == 1..MAX_HEIGHT+1
+
+\* the set of all peer ids the node can receive a message from
+AllPeerIds == CORRECT \union FAULTY
+
+Blocks == [height: Heights, wellFormed: BOOLEAN, lastCommit: BOOLEAN]
```

Comment:
what is the semantics of `lastCommit`?

Reply:
We want to model the case where lastCommit is invalid, for example that it does not have enough voting power for the same blockID. This is important for block verification logic to determine correctly a faulty peer.

Reply:
I have introduced LastCommit record and clarified semantics.